plugin/trino-hive/src/test/java/io/trino/plugin/hive/coercions/TestDecimalCoercers.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.coercions;
+
+import io.trino.spi.block.Block;
+import io.trino.spi.type.BigintType;
+import io.trino.spi.type.DecimalType;
+import io.trino.spi.type.Decimals;
+import io.trino.spi.type.IntegerType;
+import io.trino.spi.type.SmallintType;
+import io.trino.spi.type.TinyintType;
+import io.trino.spi.type.Type;
+import org.testng.annotations.Test;
+
+import java.math.BigDecimal;
+
+import static io.trino.plugin.hive.HiveTimestampPrecision.NANOSECONDS;
+import static io.trino.plugin.hive.HiveType.toHiveType;
+import static io.trino.plugin.hive.coercions.CoercionUtils.createCoercer;
+import static io.trino.spi.predicate.Utils.blockToNativeValue;
+import static io.trino.spi.predicate.Utils.nativeValueToBlock;
+import static io.trino.type.InternalTypeManager.TESTING_TYPE_MANAGER;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestDecimalCoercers
+{
+    @Test
+    public void testLongDecimalToTinyIntCoercion()
+    {
+        assertDecimalToIntCoercion(DecimalType.createDecimalType(30, 12), Decimals.valueOf(new BigDecimal("12120000000000")), TinyintType.TINYINT, 12L);
+    }
+
+    @Test
+    public void testLongDecimalToTinyIntCoercionReturnsNullWhenValueIsTooBigForTinyInt()
```

Comment:
Can we rename the method as `testLongDecimalToTinyIntBeyondRange` or something similar ?